x/staking/src/keys.rs

```diff
@@ -0,0 +1,19 @@
+use gears::types::address::BaseAddress;
+
+pub fn length_prefixed_addr_key<const PREFIX: u8>(addr: &BaseAddress<PREFIX>) -> Vec<u8> {
```

Comment:
What issue with usage of `AccAdress` or `ValAdress`?

Reply:
This handler is universal for any type of account addresses and reused from the routine [`length_prefixed_addr_pair_key`](https://github.com/rumos-io/gears/pull/105/files#diff-8251c004799364f46846eb7e3d366412d1be013eb3d61563fe57a52956f553a2R11). I used it twice in the routine: for an AccAddress and for a ValAddress. And if I'm correct, I used it once in the other method. 

The better question is: 
Why do you use generics in routine `length_prefixed_addr_pair_key`? 

I'm answering that the sdk has many methods for concrete types and it's a mess of code. This methods create keys for stores. 

I can convert the method `length_prefixed_addr_pair_key` to another signature and I'll rename it. It makes sense for now. I don't want to change the `length_prefixed_addr_key` because it is very simple and straightforward routine. 

@NYBACHOK What do you think about it?

Reply:
I was only curious about usage of this method. If you say that you reuse this method for 2 different addresses, then see no issue with keeping this one method with generic.

Reply:
I think you've found a flaw in the way we handle addresses in Gears. It was not the intention that users could access the `BaseAddress`. In future we'll change the gears implementation so that you can't. 
Do you need the flexibility here? Where it's used the first address is a `ValAddress` and the second is an `AccAddress`. If you need the flexibility then perhaps pass in the raw bytes or define your own enum. Perhaps in future we''ll have an `Address` trait in gears which you could use.